src/Chapter2.hs

```diff
@@ -348,9 +361,13 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf l
+  | null l = l
+  | otherwise = subList 0 (halfIndex (length l)) l
+  where
+    halfIndex :: Int -> Int
+    halfIndex len = (len `div` 2) - 1
```

Comment:
Note, that as `l` is in scope in here, you don't need to pass the `len` argument in here.
Instead, you can have something like this:

```haskell
halfIndex :: Int
halfIndex = (length l  `div` 2) - 1
```